packages/pulse-core/tests/state.test.ts

```diff
@@ -1,8 +1,245 @@
 import Pulse from '../lib';
 
 const App = new Pulse();
-const MyState = App.State(true);
+const BooleanState = App.State<boolean>(true);
+const StringState = App.State<string>('Hello Pulse!');
+const ObjectState = App.State<{ days: Days }>({ days: { monday: true, wednesday: true } });
+const NullState = App.State(null);
 
-test('State can be created', () => {
-  expect(MyState.value).toBe(true);
+interface Days {
+  monday?: boolean;
+  tuesday?: boolean;
+  wednesday?: boolean;
+  thursday?: boolean;
+  friday?: boolean;
+  saturday?: boolean;
+  sunday?: boolean;
+}
```

Comment:
It would be ideal to move this to the bottom of the file, or potentially in a test utilities file, to keep each .test.ts file completely dedicated to test logic.